src/timeline/widgets/VideoItem.cc

```diff
@@ -66,13 +66,15 @@ VideoItem::VideoItem(QSharedPointer<MatrixClient> client,
 
 VideoItem::VideoItem(QSharedPointer<MatrixClient> client,
                      const QString &url,
+                     const QSharedPointer<QIODevice> data,
                      const QString &filename,
                      QWidget *parent)
   : QWidget(parent)
   , url_{url}
   , text_{QFileInfo(filename).fileName()}
```

Comment:
For `VideoItem`, `AudioItem`, and `FileItem`, `filename` does actually identify a path on disk. Only `ImageItem`'s doesn't. This is because they are only created by the user explicitly selecting them to upload. 

Once they can be uploaded through other means (copy-paste), then I think it would make sense to modify this (similar to how `ImageItem` is now).

Reply:
If they need a uniform interface, then the interface should be semantically uniform. That means no inconsistent special meanings. `filename` should be standardized as purely UI.

Reply:
I'm also not a fan of this but it could be handled in a different PR.

Reply:
I also don't like it, but I thought it would be out of the scope of this PR to add changes unrelated to image pasting. I can definitely tackle this after this PR is merged.

Reply:
This PR changes the pervasive `filename` argument to file-ish widget constructors from "always a full path on disk" to not. IMO is is therefore in scope; inconsistency here is a landmine.

Reply:
I merged it before addressing the `filename` issue because I didn't want to drag this PR any further with refactoring. Although it is related, I think it would be better to have a more focused PR without the fear of moving too much out of scope.